### PR TITLE
[SYCL][NFC] Drop Gen9 detection from E2E tests

### PR DESCRIPTION
**Pull request by AlexeySachkov**
Gen9 HW is not officially supported anymore by our product, we don't have such machines in our CI and therefore it doesn't make sense to keep those legacy LIT features and their usage.

### CODE REVIEW

**Review by sarnex — APPROVED** [review]
esimd lgtm

**Review by sarnex — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]

File: sycl/test-e2e/BFloat16/bfloat16_example.cpp [hunk 1]
```diff
@@ -13,16 +13,16 @@
 // RUN: %clangxx -fsycl -fsycl-targets=spir64 %s -o %t.out
 // RUN: %{run} %t.out
 
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend "-device gen9" %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend "-device gen12lp" %s -o %t.out
 // RUN: %{run} %t.out
```
Comment (steffenlarsen) [inline sycl/test-e2e/BFloat16/bfloat16_example.cpp:17]: Seems to me like this is GPU AOT, so it should be limited to only run on GPU.
```suggestion
// RUN: %if gpu %{ %{run} %t.out %}
```

File: sycl/test-e2e/BFloat16/bfloat16_example_cpu.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+///
+/// Check if bfloat16 example works using fallback libraries AOT compiled for
+/// CPU.
+///
+
+// REQUIRES: opencl-aot, ocloc, gpu-intel-gen12, cpu
+
+// RUN: %clangxx -fsycl -fsycl-targets=spir64,spir64_gen -Xsycl-target-backend=spir64_gen "-device pvc" %s -o %t.out
```
Comment (AlexeySachkov) [inline sycl/test-e2e/BFloat16/bfloat16_example_cpu.cpp:8]: PVC is not available on Windows, so this test will fail on compilation.

File: sycl/test-e2e/BFloat16/bfloat16_example_cpu.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// RUN: %{run} %t.out
+
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device pvc" %s -o %t.out
+// RUN: %{run} %t.out
```
Comment (AlexeySachkov) [inline sycl/test-e2e/BFloat16/bfloat16_example_cpu.cpp:12]: `RUN` line is not guarded by `if`, so we will attempt to launch AOT-compiled code targeting PVC on a gen12 GPU. That will work thanks to AOT -> JIT fallback, but I don't think that we should do that

**Comment by AlexeySachkov** [comment]
I cannot approve my own PR, but my comments about recent changes were addressed, so I'm fine with merging this.

**Comment by steffenlarsen** [comment]
Primary changes since approvals were that the bfloat16_example has been split into separate files to avoid simplify the cpu/gpu conditional runs/builds.

@sarnex | @kbenzie | @againull - Would you like to re-review?

**Comment by steffenlarsen** [comment]
Thank you, @sarnex ! Since there were no other reactions, we will proceed with the merge. Please feel free to do a post-commit review and we will address it in a separate patch.

**Comment by AlexeySachkov** [comment]
This PR broke post-commit on AMD, the fix is in #15533

**Pull request merged**